### DISCUSSION TITLE
Кнопка переключения видимости дополнительных полей в отчете

### DISCUSSION THREAD

**Discussion by koloskovy** [fralx/LimeReport#465] [Q&A] [answered]
Я собрал проект в QT Creator 4.6.2 LimeReport  на Astra Linux 1.6 в Lime Report Disigner я создал отчет некоторые поля этого отчета скрыты и должны показываться при нажатии кнопки. Я написал скрипт в lime report designer который отвечает за данный функционал, но сделать кнопку, которая по нажатию переключала бы видимость дополнительных полей не удалось. Как решить данную проблему? Я уже отдельно в инспектора скриптов создал диалог с кнопкой и в своей скрипте это учел, но ничего не появляется. Текстовое поле которое выполняет роль переключателя не реагирует на нажатия остаётся только надписью. Если в самом скрипте вручную менять у одной переменной false на true и наоборот дополнительные поля либо появляются либо скрываются в Сгенерироанном отчёте но мне надо чтобы это происходило по нажатию кнопки ведь пользователь не должен вручную править скрипт Подскажите пожалуйста есть ли способ выводить кнопку в сгенерированном отчете по нажатию на которую дополнительные поля будут либо показываться, либо скрываться Вот мой код во вкладке Скрипт в Lime Report Designer var showAdditionalFields = false;
var toggleCounter = 0;

function toggleFields() {
    toggleCounter++;
    if (toggleCounter % 2 === 0) {
        showAdditionalFields = !showAdditionalFields;
    }
    return getButtonText();
}

function getButtonText() {
    if (toggleCounter % 2 === 0) {
        return showAdditionalFields ? "Дополнительные поля показаны" : "Дополнительные поля скрыты";
    } else {
        return "Нажмите для применения изменений";
    }
}

function isFieldVisible(fieldName) {
    var basicFields = ["actual_time", "delivery_time", "mon_obj_id"];
    if (basicFields.indexOf(fieldName) !== -1) {
        return true;
    }
    return showAdditionalFields;
}

**Comment by fralx** [comment, marked as answer]
Если я правильно понял ваш вопрос то вот пример как можно в диалоге отчета скрывать и показывать поля
[ShowFieldsDialog.lrxml.zip](https://github.com/user-attachments/files/16198069/ShowFieldsDialog.lrxml.zip)

**Comment by koloskovy** [comment]
Да, вы все правильно поняли. Ваш пример, то что надо. Спасибо вам огромное!!!

**Comment by koloskovy** [comment]
Подскажите пожалуйста что в свойствах диалога. Создал в QT Disigner свой диалог импортировал его в LimeReport сделал все по аналогии с вашим примером, при нажатии кнопки выбора на диалоге с отмеченным чекбоксом диалоговое окно не убирается и отчет не генерируется. С вашим примером все работает. Не могу понять что с моим диалоговым окном, может где-то в свойствах что-то в QT Designer надо было указать. Не подскажете?

**Comment by fralx** [comment]
В дизайнере диалогов надо соединить сигнал кнопки (clicked) со слотом формы (accept).  
![SignalSlot](https://github.com/user-attachments/assets/7dbef36e-9c79-4b5e-9c48-5ff30358c964)
 Перейти в режим редактирования сигналов и слотов, кликнуть на кнопку и с зажатой левой кнопкой вытянуть соединение и соединить с формой в диалоге выбрать clicked и accept
![SignalSlotEditor](https://github.com/user-attachments/assets/04c82628-995c-4756-9000-3d5288ecea2f)

**Comment by koloskovy** [comment]
Спасибо. А у меня проект собрался без редактора диалогов, я его создавал в QT Designer
![Снимок](https://github.com/user-attachments/assets/72edbb19-415b-408f-b069-88de221dca68)

**Comment by fralx** [comment]
У  вас получилось диалог подключить к отчету? 
Если да то в QT Designer нужно описанные действия выполнить

**Comment by koloskovy** [comment]
Да, конечно в QT Designer все получилось. Спасибо вам! Но просто интересно, где при сборке я что-то упустил, что собранный LimeReport без редактора диалогов, не такой как у вас на скринах?

**Reply by fralx** [reply]
Если CMAKE использовали или статическую сборку то дизайнер не соберется, наиболее полная сборка через  QMAKE

**Reply by koloskovy** [reply]
Все понятно. Да собирал через CMAKE. Спасибо.
